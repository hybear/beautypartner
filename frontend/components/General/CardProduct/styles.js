import styled from 'styled-components';
import Text from '../Text';

export const CardStyles = styled.div`
  background: white;
  border: 1px solid ${({ theme }) => theme.color.lightGrey};
  display: flex;
  flex-direction: column;
  position: relative;
  transition: all 0.2s;
  &:hover {
    border: 1px solid white;
    box-shadow: ${({ theme }) => theme.misc.bs};
    transform: translateY(-5px);
  }

  .buttonList {
    display: grid;
    width: 100%;
    border-top: 1px solid ${({ theme }) => theme.color.lightgrey};
    grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
    grid-gap: 1px;
    justify-content: space-between;
    background: ${({ theme }) => theme.color.lightgrey};
    & > * {
      background: white;
      border: 0;
      font-size: 1em;
      padding: 1em;
      text-align: center;
    }
  }
`;

export const Image = styled.img`
  width: 100%;
  height: 220px;
  object-fit: contain;
  padding: 15px;
`;

export const Title = styled.h3`
  text-align: center;
  width: 100%;
  margin-bottom: 10px;
  display: block;
  line-height: 1.1;
  font-size: 1em;
  overflow: hidden;
  height: 1em;
  text-align: center;
  color: ${({ theme }) => theme.color.black};
  padding: 0 1rem;
`;

export const Description = styled(Text)`
  margin-bottom: 10px;
  max-height: 50px;
  font-size: 0.875em;
  color: ${({ theme }) => theme.color.grey};
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
  -webkit-line-clamp: 2;
`;

export const AddToCart = styled.div`
  margin: 10px 10px 15px;
  opacity: ${({ hover }) => (hover ? '1' : '0')};
  transition: all 0.2s;
`;

export const PriceTag = styled.span`
  background: ${({ theme }) => theme.color.primary};
  color: ${({ theme }) => theme.color.white};
  padding: 5px;
  line-height: 1;
  font-size: 1.25em;
  display: inline-block;
  position: absolute;
  top: 0;
  right: 0;
`;

export const Secundary = styled.button``;
