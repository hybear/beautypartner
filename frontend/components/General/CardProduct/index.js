import React, { useState } from 'react';
import { CardStyles, Title, Image, PriceTag, Description, AddToCart } from './styles';
import AddBtn from '../CartAddItem';
import formatMoney from '../../../lib/formatMoney';

const CardProduct = props => {
  const [Hover, setHover] = useState(false);

  const { item } = props;

  return (
    <CardStyles onMouseEnter={() => setHover(!Hover)} onMouseLeave={() => setHover(!Hover)}>
      {/* <Link href={{
                        pathname: `/p/[name]/[id]`
                    }} as={{
                        pathname: `/p/${item.title.replace(/ /g, '-')}/${item.id}`
                    }}>
                    <a> */}
      <Image src={item.image} alt={item.title} />
      <Title>{item.title}</Title>
      <PriceTag>{formatMoney(item.listPrice)}</PriceTag>
      <Description center>{item.description}</Description>
      {/* </a>
                    </Link> */}
      <AddToCart hover={Hover}>
        <AddBtn primary id={item.id} />
      </AddToCart>
      {/* {user && (
        <>
          {user.permissions.map(e => e  != 'USER') && (
            <div className="buttonList">
              {user.permissions.map(e => e == 'ADMIN' || e == 'ITEMUPDATE') && (
                <Link
                  href={{
                    pathname: 'update',
                    query: { id: item.id },
                  }}
                >
                  <a className="buttonList__edit">Edit</a>
                </Link>
              )}
              {user.permissions.map(e => e == 'ADMIN' || e == 'ITEMDELETE') && (
                <DeleteItem id={item.id}>Delete</DeleteItem>
              )}
            </div>
          )}
        </>
      )} */}
    </CardStyles>
  );
};

export default CardProduct;
