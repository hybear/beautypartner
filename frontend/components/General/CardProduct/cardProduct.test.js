import CardProduct from './index'

const fakeItem = {
    id: 'ABC123',
    title: 'A cool product',
    listPrice: 5000,
    bestPrice: 5000,
    description: "this is really cool",
    image: 'parfum.jpg',
    largeImage: 'largeparfum.jpg'
}

describe('<Item/>', () => {
    it('renders and displays Price properly', ()=>{
        const wrapper = shallow(<CardProduct item={fakeItem} />);
        const PriceTag = wrapper.find('styles__PriceTag');
        expect(PriceTag.children().text()).toBe('$50');
    })
    it('renders and displays Image properly', ()=>{
        const wrapper = shallow(<CardProduct item={fakeItem} />);
        const img = wrapper.find('styles__Image')
        expect(img.props().src).toBe(fakeItem.image);
    })
})

describe('<Item/>', () => {
    it('renders and matches the snapshot', () => {
        const wrapper = shallow(<CardProduct item={fakeItem} />);
        expect(wrapper).toMatchSnapshot();
    })
})