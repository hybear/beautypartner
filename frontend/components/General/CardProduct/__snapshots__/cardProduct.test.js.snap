// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<styles__CardStyles
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Link
    as={
      Object {
        "pathname": "/p/A-cool-product/ABC123",
      }
    }
    href={
      Object {
        "pathname": "/p/[name]/[id]",
      }
    }
  >
    <a>
      <styles__Image
        alt="A cool product"
        src="parfum.jpg"
      />
      <styles__Title>
        A cool product
      </styles__Title>
      <styles__PriceTag>
        $50
      </styles__PriceTag>
      <styles__Description
        center={true}
      >
        this is really cool
      </styles__Description>
    </a>
  </Link>
  <styles__AddToCart
    hover={false}
  >
    <AddToCart
      id="ABC123"
    />
  </styles__AddToCart>
</styles__CardStyles>
`;
